include/dav1d/picture.rs

```diff
@@ -320,7 +320,7 @@ impl From<Rav1dPicAllocator> for Dav1dPicAllocator {
 }
 
 impl Rav1dPicAllocator {
-    pub unsafe fn alloc_picture(&mut self, p: *mut Rav1dPicture) -> Rav1dResult {
+    pub unsafe fn alloc_picture(&self, p: *mut Rav1dPicture) -> Rav1dResult {
```

Comment:
This should come along with a doc comment saying internal access to `Rav1dPicAllocator::cookie` (a `*mut c_void`) must be thread-safe.  Otherwise the `&self` here would be UB.

Reply:
A raw pointer is not Send or Sync, so we're going to have to deal with the thread safety of this structure later anyway. I'll have to look at what the existing code does, it may be a moot point and not actually accessed from multiple threads.

What is true even without thread safety is that the callbacks must not be re-entrant, or if they are they must use the cookie in a manner that is reentrant safe.

Reply:
Handled in #680